src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -135,12 +142,124 @@ public SimpleInterval(final String str){
         this.end = end;
     }
 
-    /**
+     /**
+      * Given an interval query string and a sequence dictionary, determine if the query string can be
+      * resolved as a valid interval query against more than one contig in the dictionary, i.e., more than
+      * one of:
+      *
+      *     prefix
+      *     prefix:nnn
+      *     prefix:nnn+
+      *     prefix:nnn-nnn
+      *
+      * and return the list of all possible interpretations (there can never be more than 2). Note that for
+      * an ambiguity to exist, the query string must contain at least one colon.
+      *
+      * @param intervalQueryString
+      * @param sequenceDictionary
+      * @return List<SimpleInterval> containing 0, 1, or 2 valid interpretations of {code queryString} given
+      * {@code sequenceDictionary}. If the list is empty, the query doesn't match any contig in the sequence
+      * dictionary. If the list contains more than one interval, the query string is ambiguous and should be
+      * rejected. If the list contains a single interval, the query is unambiguous and can be safely used to
+      * conduct a query.
+      * @thows IllegalArgumentException if the query only matches a single contig in the dictionary, but the start
+      * and end positions are not valis
+      * @throws NumberFormatException if the query only matches a single contig in the dictionary, but the query
+      * interval paramaters (start, end) cannot be parsed
+      */
+     public static List<SimpleInterval> getResolvedIntervals(
```

Comment:
Can this be moved to `IntervalUtils`? Have this huge utility method directly in `SimpleInterval` like this is a little gross...  (if it would be non-trivial we can move it in a separate PR).

Reply:
Done.